controllers/agentpool_controller_autoscaling.go

```diff
@@ -151,16 +151,33 @@ func (r *AgentPoolReconciler) scaleAgentDeployment(ctx context.Context, ap *agen
 	return r.Client.Update(ctx, &deployment)
 }
 
-// coolDownSecondsRemaining returns the remaining seconds in the Cool Down stage.
+// cooldownSecondsRemaining(currentReplica returns the remaining seconds in the Cool Down stage.
 // A negative value indicates expired Cool Down.
-func (a *agentPoolInstance) coolDownSecondsRemaining() int {
-	if s := a.instance.Status.AgentDeploymentAutoscalingStatus; s != nil && s.LastScalingEvent != nil {
-		lastScalingEventSeconds := int(time.Since(s.LastScalingEvent.Time).Seconds())
-		cooldownPeriodSeconds := int(*a.instance.Spec.AgentDeploymentAutoscaling.CooldownPeriodSeconds)
-		return cooldownPeriodSeconds - lastScalingEventSeconds
+func (a *agentPoolInstance) cooldownSecondsRemaining(currentReplicas, desiredReplicas int32) int {
+	status := a.instance.Status.AgentDeploymentAutoscalingStatus
+	if status == nil || status.LastScalingEvent == nil {
+		return -1
 	}
 
-	return -1
+	cooldownPeriodSeconds := int(*a.instance.Spec.AgentDeploymentAutoscaling.CooldownPeriodSeconds)
+
+	cooldownPeriod := a.instance.Spec.AgentDeploymentAutoscaling.CooldownPeriod
+	if cooldownPeriod != nil {
+		if v := cooldownPeriod.ScaleDownSeconds; v != nil {
+			cooldownPeriodSeconds = int(*v)
+			a.log.Info("Reconcile Agent Autoscaling", "msg", fmt.Sprintf("Agents scaling down, using configured scale down period: %v", cooldownPeriodSeconds))
+		}
+
+		if v := cooldownPeriod.ScaleUpSeconds; v != nil {
+			if desiredReplicas > currentReplicas {
+				cooldownPeriodSeconds = int(*v)
+				a.log.Info("Reconcile Agent Autoscaling", "msg", fmt.Sprintf("Agents scaling up, using configured scale down period: %v", cooldownPeriodSeconds))
+			}
+		}
+	}
+
+	lastScalingEventSeconds := int(time.Since(status.LastScalingEvent.Time).Seconds())
+	return cooldownPeriodSeconds - lastScalingEventSeconds
```

Comment:
To make the logic here clear:
1. Default to using the value from `cooldownPeriodSeconds` as the cooldown period. 
2. If `cooldown.scaleDownSeconds` is set, use that value as the cooldown period. 
3. If `cooldown.scaleUpSeconds` is set, and we are scaling up, use that value as the cooldown period.